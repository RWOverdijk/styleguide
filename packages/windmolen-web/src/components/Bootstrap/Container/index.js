import React from 'react';
import styled from 'styled-components';
import { px } from '../../../utils';
import { gridTheme } from '../../../globals';
import { media, Container as BootstrapContainer } from 'styled-bootstrap-grid';

const CONTAINER_SMALL = 760 + (gridTheme.container.padding * 2);
const CONTAINER_MEDIUM = 960 + (gridTheme.container.padding * 2);

const Container = styled(BootstrapContainer)`
  ${props => props.small && `max-width: ${px(CONTAINER_SMALL)};`}
  ${props => props.medium && `max-width: ${px(CONTAINER_MEDIUM)};`}

  ${props => (!props.small && !props.medium) && media.giant`
    max-width: 1200px;
  `}
  }
`;

export default Container;
