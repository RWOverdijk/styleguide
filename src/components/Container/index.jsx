import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import './style.css';

class Container extends PureComponent {
  getClass() {
    let className = 'container';

    if (this.props.fluid) {
      className += '-fluid';
    }

    return className;
  }
  render() {
    return <div className={this.getClass()}>{this.props.children}</div>;
  }
}

Container.propTypes = {
  fluid: PropTypes.bool
};

export default Container;
