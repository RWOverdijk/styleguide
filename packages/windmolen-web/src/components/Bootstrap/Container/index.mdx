---
name: Container
menu: Bootstrap
route: /Bootstrap/Container
---

import { Playground, PropsTable } from 'docz'
import Wrapper from '../../Wrapper';
import Container from './';
import Col from '../Col';
import Row from '../Row';

# Container

## Container
<Playground wrapper={Wrapper}>
  <Container>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et mi
    interdum, iaculis est id, maximus elit. Maecenas at sem scelerisque,
    vestibulum nibh vel, feugiat tortor. Pellentesque consectetur eros eu lacus
    pretium accumsan. Nulla quam urna, imperdiet sed nibh sit amet, consequat
    vehicula dolor. Vestibulum ante ipsum primis in faucibus orci luctus et
    ultrices posuere cubilia Curae; Maecenas ut viverra dui. Ut in auctor est.
    Nam molestie at odio eget fermentum. Suspendisse potenti. Etiam venenatis
    posuere nulla, a feugiat est molestie id. Integer purus tellus, placerat
    eget ligula ut, ornare mollis mauris. Suspendisse lectus libero, commodo eu
    pretium in, mollis sit amet ex. Donec malesuada feugiat turpis eget
    lacinia.
  </Container>
</Playground>

## Container medium
<Playground wrapper={Wrapper}>
  <Container medium>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et mi
    interdum, iaculis est id, maximus elit. Maecenas at sem scelerisque,
    vestibulum nibh vel, feugiat tortor. Pellentesque consectetur eros eu lacus
    pretium accumsan. Nulla quam urna, imperdiet sed nibh sit amet, consequat
    vehicula dolor. Vestibulum ante ipsum primis in faucibus orci luctus et
    ultrices posuere cubilia Curae; Maecenas ut viverra dui. Ut in auctor est.
    Nam molestie at odio eget fermentum. Suspendisse potenti. Etiam venenatis
    posuere nulla, a feugiat est molestie id. Integer purus tellus, placerat
    eget ligula ut, ornare mollis mauris. Suspendisse lectus libero, commodo eu
    pretium in, mollis sit amet ex. Donec malesuada feugiat turpis eget
    lacinia.
  </Container>
</Playground>

## Container small
<Playground wrapper={Wrapper}>
  <Container small>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et mi
    interdum, iaculis est id, maximus elit. Maecenas at sem scelerisque,
    vestibulum nibh vel, feugiat tortor. Pellentesque consectetur eros eu lacus
    pretium accumsan. Nulla quam urna, imperdiet sed nibh sit amet, consequat
    vehicula dolor. Vestibulum ante ipsum primis in faucibus orci luctus et
    ultrices posuere cubilia Curae; Maecenas ut viverra dui. Ut in auctor est.
    Nam molestie at odio eget fermentum. Suspendisse potenti. Etiam venenatis
    posuere nulla, a feugiat est molestie id. Integer purus tellus, placerat
    eget ligula ut, ornare mollis mauris. Suspendisse lectus libero, commodo eu
    pretium in, mollis sit amet ex. Donec malesuada feugiat turpis eget
    lacinia.
  </Container>
</Playground>

## Container + Columns
<Playground wrapper={Wrapper}>
  <Container>
    <Row>
      <Col md={6}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et mi
        interdum, iaculis est id, maximus elit. Maecenas at sem scelerisque,
        vestibulum nibh vel, feugiat tortor. Pellentesque consectetur eros eu lacus
        pretium accumsan. Nulla quam urna, imperdiet sed nibh sit amet, consequat
        vehicula dolor. Vestibulum ante ipsum primis in faucibus orci luctus et
        ultrices posuere cubilia Curae; Maecenas ut viverra dui. Ut in auctor est.
        Nam molestie at odio eget fermentum. Suspendisse potenti. Etiam venenatis
        posuere nulla, a feugiat est molestie id. Integer purus tellus, placerat
        eget ligula ut, ornare mollis mauris. Suspendisse lectus libero, commodo eu
        pretium in, mollis sit amet ex. Donec malesuada feugiat turpis eget
        lacinia.
      </Col>
      <Col md={6}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et mi
        interdum, iaculis est id, maximus elit. Maecenas at sem scelerisque,
        vestibulum nibh vel, feugiat tortor. Pellentesque consectetur eros eu lacus
        pretium accumsan. Nulla quam urna, imperdiet sed nibh sit amet, consequat
        vehicula dolor. Vestibulum ante ipsum primis in faucibus orci luctus et
        ultrices posuere cubilia Curae; Maecenas ut viverra dui. Ut in auctor est.
        Nam molestie at odio eget fermentum. Suspendisse potenti. Etiam venenatis
        posuere nulla, a feugiat est molestie id. Integer purus tellus, placerat
        eget ligula ut, ornare mollis mauris. Suspendisse lectus libero, commodo eu
        pretium in, mollis sit amet ex. Donec malesuada feugiat turpis eget
        lacinia.
      </Col>
    </Row>
  </Container>
</Playground>

## Container + Column with offset
<Playground wrapper={Wrapper}>
  <Container>
    <Row>
      <Col md={10} mdOffset={1}>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec et mi
        interdum, iaculis est id, maximus elit. Maecenas at sem scelerisque,
        vestibulum nibh vel, feugiat tortor. Pellentesque consectetur eros eu lacus
        pretium accumsan. Nulla quam urna, imperdiet sed nibh sit amet, consequat
        vehicula dolor. Vestibulum ante ipsum primis in faucibus orci luctus et
        ultrices posuere cubilia Curae; Maecenas ut viverra dui. Ut in auctor est.
        Nam molestie at odio eget fermentum. Suspendisse potenti. Etiam venenatis
        posuere nulla, a feugiat est molestie id. Integer purus tellus, placerat
        eget ligula ut, ornare mollis mauris. Suspendisse lectus libero, commodo eu
        pretium in, mollis sit amet ex. Donec malesuada feugiat turpis eget
        lacinia.
      </Col>
    </Row>
  </Container>
</Playground>
