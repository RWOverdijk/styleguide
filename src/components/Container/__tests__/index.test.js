import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../';

describe("<Container>", () => {
  it("renders Container", () => {
    const tree = renderer.create(
      <Container>My Container</Container>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders fluid Container", () => {
    const tree = renderer.create(
      <Container fluid>My Container</Container>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <Container className="my-class">My Container</Container>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <Container data-id="0">My Container</Container>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
