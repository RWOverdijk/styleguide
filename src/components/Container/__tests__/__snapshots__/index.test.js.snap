// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container> renders Container 1`] = `
<div
  className="container"
>
  My Container
</div>
`;

exports[`<Container> renders fluid Container 1`] = `
<div
  className="container-fluid"
>
  My Container
</div>
`;

exports[`<Container> renders with specific attributes 1`] = `
<div
  className="container"
  data-id="0"
>
  My Container
</div>
`;

exports[`<Container> renders with specific className 1`] = `
<div
  className="container my-class"
>
  My Container
</div>
`;
